src/eureka/S3_data_reduction/wfc3.py

```diff
@@ -409,30 +433,50 @@ def read(filename, data, meta, log):
                             -1*data.attrs['shdr']['LTV1']]]
         data.attrs['exptime'] = data.attrs['mhdr']['EXPTIME']
         flux_units = data.attrs['shdr']['BUNIT']
+        temp_scandir = meta.scandir[filename == meta.segment_list]
 
         # Determine if we are using IMA or FLT files
-        if filename.endswith('flt.fits'):
-            # FLT files subtract first from last, 2 reads
-            meta.nreads = 1
+        if not filename.endswith('flt.fits'):
+            meta.local_nreads = data.attrs['shdr']['SAMPNUM']
         else:
-            meta.nreads = data.attrs['shdr']['SAMPNUM']
-
-        sci = np.zeros((meta.nreads, meta.ny, meta.nx))  # Flux
-        err = np.zeros((meta.nreads, meta.ny, meta.nx))  # Error
-        dq = np.zeros((meta.nreads, meta.ny, meta.nx))  # Flags
-        jd = []
-        for j, rd in enumerate(range(meta.nreads, 0, -1)):
+            meta.local_nreads = 1
+
+        start = 0
+        end = meta.local_nreads
+        if meta.local_nreads != meta.nreads_full:
+            # File had one fewer read, figure out which side to pad with NaN
+            if temp_scandir == 0:
+                start = 0
+                end = meta.local_nreads
+            else:
+                start = 1
+                end = meta.local_nreads+1
+
+        sci = np.nan*np.ones((meta.nreads_full, meta.ny, meta.nx))  # Flux
+        err = np.nan*np.ones((meta.nreads_full, meta.ny, meta.nx))  # Error
+        dq = np.nan*np.ones((meta.nreads_full, meta.ny, meta.nx))  # Flags
+        jd = np.nan*np.ones(meta.nreads_full)
```

Comment:
I think it's cleaner to use `np.full(shape, np.nan)`

Reply:
Oh neat, I've never seen that function! Will do